src/doc/stdmetadata.rst

```diff
@@ -284,14 +284,94 @@ The following metadata items are specific to photos or captured images.
 
     For captured or scanned image, the model of the camera or scanner.
 
+.. option:: "Focus" : float
+
+    The camera's focus distance of the captured image.
+
```

Comment:
I really believe that before we add to this list of "standard OIIO metadata", we should have an open discussion about what to add, and why. And I'd rather not do it item by item, but instead, propose a general rule or philosophy about the naming conventions. For example, maybe we should just decide that all of the SMPTE standard names be adopted. Or maybe we should vow to always follow OpenEXR's names and set of standard attributes. Or maybe something else. And regardless, we need to think about exactly how, or if, we should transition from any of the contradictory names we've been using and how to handle it going forward.

You've waded into a longstanding open question about policy and design philosophy, and I don't think it can be solved by a PR that makes some specific changes, not others, and doesn't explain itself at all as to what design its trying to implement. Start with that -- a proposal for how we should approach the metadata problem, and then the stakeholders can collectively decide what we want to do about it, and after that, an implementation that embodies the decision should be straightforward.

Reply:
Adopting metadata is essentially buying in to a model of the camera, and of image processing, that the metadata describes. To take the example of camera correlated color temperature (CCT) and tint, please see the discussion of how cameraCCTSetting, cameraTintSetting and cameraColorBalance relate to each other in ImfStandardAttributes.h, and the distinction between forensic and prescriptive metadata.

As the author of the new definitions that were added in 2023 I am keenly interested in seeing these used, but used effectively, and that means doing the philosophical review Larry mentions. I am biased, but I think OpenEXR has done the best job so far in precisely defining metadata semantics, types, and value constraints; but to reconcile their use with OpenImageIO's pan-image-format model, we need to not only look at other formats' definitions, but also to the elephant in the room: Nuke 15's really awful switch to strings for numeric attributes, carrying both the numeric value but also, appended as characters, the units. Eventually I hope to bring them back into the fold, but for those files they've already generated, what should we do? Reject entirely? Convert to canonical form? And what about an image that is read, flopped (say), and rewritten? Do we write the mangled values or canonicalized ones?

This thread feels somehow frantically urgent to the PR submitter; or is that just my take on it?

Reply:
An additional unsettled item for me is what the best thing to do when we see that two widely adopted metadata conventions have different names for things. For example, when reading from formats that support Exif, we name the metadata "Exif:NAME" where the NAME is taken straight from the word the Exif spec uses to describe the item. What do we do when a particular piece of Exif data seems to *exactly* correspond to an item described by OpenEXR/SMPTE (which at the very least won't have the "Exif:" namespace prefix, and may additionally not have the same name stem).